stdlib/public/core/String.swift

```diff
@@ -1246,3 +1246,18 @@ extension String : LosslessStringConvertible {
     self = content
   }
 }
+
+extension String {
+  //FIXME: This should not be public, this is just a workaround for the tests
+  public func _withNormalizedUTF8Iterator<ReturnType>(_ f: (_NormalizedUTF8CodeUnitIterator) -> ReturnType) -> ReturnType {
+    if _fastPath(_guts.isFastUTF8) {
```

Comment:
Assuming you want to keep the type public, we already have this here https://github.com/milseman/swift/blob/utf8string/stdlib/public/core/StringComparison.swift#L39, so the implementation could be like:

`return try _slicedGuts.withNFCCodeUnits(f)`